src/bundle/DependencyInjection/Factory/Security/TwoFactorFactory.php

```diff
@@ -110,6 +122,46 @@ public function create(ContainerBuilder $container, $id, $config, $userProvider,
         return [$providerId, $listenerId, $defaultEntryPoint];
     }
 
+    public function createAuthenticator(ContainerBuilder $container, string $id, array $config, string $userProviderId): string
+    {
+        $csrfTokenManagerId = $this->twoFactorServicesFactory->getCsrfTokenManagerId($config); // TODO this has to go somewhere
+        $twoFactorFirewallConfigId = $this->twoFactorServicesFactory->createTwoFactorFirewallConfig($container, $id, $config);
+        $successHandlerId = $this->twoFactorServicesFactory->createSuccessHandler($container, $id, $config, $twoFactorFirewallConfigId);
+        $failureHandlerId = $this->twoFactorServicesFactory->createFailureHandler($container, $id, $config, $twoFactorFirewallConfigId);
+        $authRequiredHandlerId = $this->twoFactorServicesFactory->createAuthenticationRequiredHandler($container, $id, $config, $twoFactorFirewallConfigId);
+        $this->twoFactorServicesFactory->createKernelExceptionListener($container, $id, $authRequiredHandlerId);
+        $this->twoFactorServicesFactory->createAccessListener($container, $id, $twoFactorFirewallConfigId);
+        $this->twoFactorServicesFactory->createProviderPreparationListener($container, $id, $config);
+
+        return $this->createAuthenticatorService(
+            $container,
+            $id,
+            $twoFactorFirewallConfigId,
+            $successHandlerId,
+            $failureHandlerId,
+            $authRequiredHandlerId
+        );
+    }
+
+    private function createAuthenticatorService(
+        ContainerBuilder $container,
+        string $firewallName,
+        string $twoFactorFirewallConfigId,
+        string $successHandlerId,
+        string $failureHandlerId,
+        string $authRequiredHandlerId
+    ): string {
+        $authenticatorId = self::AUTHENTICATOR_ID_PREFIX.$firewallName;
+        $container
+            ->setDefinition($authenticatorId, new ChildDefinition(self::AUTHENTICATOR_DEFINITION_ID))
+            ->replaceArgument(0, new Reference($twoFactorFirewallConfigId))
+            ->replaceArgument(2, new Reference($successHandlerId))
+            ->replaceArgument(3, new Reference($failureHandlerId))
+            ->replaceArgument(4, new Reference($authRequiredHandlerId));
+
+        return $authenticatorId;
+    }
```

Comment:
Technically, it could. Conceptually, I thought of `AuthenticatorTwoFactorFactory` more as the construct I use to create compatibility between different Symfony versions, rather than an actual class.